sale_start_end_dates/models/sale.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2014-2016 Akretion (http://www.akretion.com)
+# @author Alexis de Lattre <alexis.delattre@akretion.com>
+# Copyright 2016 Sodexis (http://sodexis.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+from dateutil.relativedelta import relativedelta
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    default_start_date = fields.Date(string='Default Start Date')
+    default_end_date = fields.Date(string='Default End Date')
+
+    @api.constrains('default_start_date', 'default_end_date')
+    def _check_default_start_end_dates(self):
+        if (self.default_start_date and self.default_end_date and
+                self.default_start_date > self.default_end_date):
+            raise ValidationError(
+                _("Default Start Date should be before or be the "
+                    "same as Default End Date for sale order %s")
+                % self.name)
+
+    @api.onchange('default_start_date')
+    def default_start_date_change(self):
+        if (self.default_start_date and self.default_end_date and
+                self.default_start_date > self.default_end_date):
+            self.default_end_date = self.default_start_date
+
+    @api.onchange('default_end_date')
+    def default_end_date_change(self):
+        if (self.default_start_date and self.default_end_date and
+                self.default_start_date > self.default_end_date):
+            self.default_start_date = self.default_end_date
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
```

Comment:
please, consider splitting the files for different models. It makes reading modules a lot better, as well as it quickly shows what models are used in the module by just glancing at the file names in models directory